esp-hal-common/src/analog/adc/cal_line.rs

```diff
@@ -0,0 +1,98 @@
+use core::marker::PhantomData;
+
+use crate::adc::{
+    AdcCalBasic,
+    AdcCalEfuse,
+    AdcCalScheme,
+    AdcCalSource,
+    AdcConfig,
+    Attenuation,
+    RegisterAccess,
+};
+
+/// Marker trait for ADC unit which support line fitting
+///
+/// Usually it means that reference points is stored in efuse.
+/// See also [`AdcCalLine`].
+pub trait AdcHasLineCal {}
+
+// coeff_a is actually a float number
+// it is scaled to put them into uint32_t so that the headers do not have to be
+// changed
```

Comment:
This comment also needs to be updated, and probably turned into a proper doc comment?